mustache.js

```diff
@@ -420,6 +426,19 @@
           }
         } else {
           intermediateValue = context.view[name];
+
+          /**
+           * Only checking against hasProperty, which always returns false if
+           * context.view is not an object. Deliberately omitting the check
+           * against primitiveHasOwnProperty if dot notation is not used.
+           *
+           * Consider this example:
+           * Mustache.render("{{#length}}{{length}}{{/length}}", {length: "hello"})
```

Comment:
The example above doesn't illustrate the `bug_length_property` test case because we're not making a primitive the current context -- it renders the same way with our without the `primitiveHasOwnProperty` check. TBH I'm having trouble coming up with a natural example of where the `bug_length_property` bug would manifest. Best I could come up with is `Mustache.render("The length of a football field is {{#length}}{{length}}{{/length}}", {length: "100 yards"})`, where _with_ the check it would render as "The length of a football field is 9" rather than "The length of a football field is 100 yards".